core/src/main/kotlin/de/fraunhofer/iem/kpiCalculator/core/strategy/WeightedMaximumKPICalculationStrategy.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2024 Fraunhofer IEM. All rights reserved.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ * SPDX-License-Identifier: MIT
+ * License-Filename: LICENSE
+ */
+
+package de.fraunhofer.iem.kpiCalculator.core.strategy
+
+import de.fraunhofer.iem.kpiCalculator.core.hierarchy.KpiHierarchyEdge
+import de.fraunhofer.iem.kpiCalculator.model.kpi.KpiStrategyId
+import de.fraunhofer.iem.kpiCalculator.model.kpi.hierarchy.KpiCalculationResult
+import de.fraunhofer.iem.kpiCalculator.model.kpi.hierarchy.KpiNode
+import io.github.oshai.kotlinlogging.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+
+internal object WeightedMaximumKPICalculationStrategy : BaseKpiCalculationStrategy() {
+
+    override val kpiStrategyId: KpiStrategyId
+        get() = KpiStrategyId.WEIGHTED_MAXIMUM_STRATEGY
+
+    override fun internalCalculateKpi(edges: Collection<KpiHierarchyEdge>): KpiCalculationResult {
+
+        val max = if (edges.isEmpty()) 0 else edges.maxOf { it.to.score * it.actualWeight }.toInt()
```

Comment:
is it intentional that e.g, `5.9` is converted to `5` instead of maybe a rounded value `6`?

Reply:
yep. as discussed we just opted to use .toInt() as it doesn't make a semantic difference and makes the necessary logic and code to write way easier